File: frontend/src/features/inbox/api/inboxController.ts
import api from "../../../shared/api/api";
import { ApiResponse } from "../../../shared/types/ApiResponse";
import { InboxPageDto } from "../types/InboxNotificationDto";

export const inboxController = {
    getInbox: async (skip: number, take: number): Promise<ApiResponse<InboxPageDto>> =>
        api.get(`/inbox?skip=${skip}&take=${take}`),

    getUnreadCount: async (): Promise<ApiResponse<number>> =>
        api.get(`/inbox/unread-count`),

    markRead: async (id: string): Promise<ApiResponse> =>
        api.post(`/inbox/${id}/read`, {}),

    markAllRead: async (): Promise<ApiResponse> =>
        api.post(`/inbox/read-all`, {}),

    deleteItem: async (id: string): Promise<ApiResponse> =>
        api.delete(`/inbox/${id}`),
};

File: backend/src/Operum.API/Controllers/InboxController.cs
using Microsoft.AspNetCore.Mvc;
using Operum.API.Controllers.Base;
using Operum.API.Filters;
using Operum.Service.Interfaces;

namespace Operum.API.Controllers
{
    [ApiController]
    [RequiresNotifications]
    [Route("api/[controller]")]
    public class InboxController(IInboxService inboxService) : BaseController
    {
        [HttpGet]
        public async Task<IActionResult> GetInbox([FromQuery] int skip = 0, [FromQuery] int take = 20)
        {
            return GetApiResponse(await inboxService.GetInbox(skip, take));
        }

        [HttpGet("unread-count")]
        public async Task<IActionResult> GetUnreadCount()
        {
            return GetApiResponse(await inboxService.GetUnreadCount());
        }

        [HttpPost("{id}/read")]
        public async Task<IActionResult> MarkRead([FromRoute] string id)
        {
            return GetApiResponse(await inboxService.MarkRead(id));
        }

        [HttpPost("read-all")]
        public async Task<IActionResult> MarkAllRead()
        {
            return GetApiResponse(await inboxService.MarkAllRead());
        }

        [HttpDelete("{id}")]
        public async Task<IActionResult> Delete([FromRoute] string id)
        {
            return GetApiResponse(await inboxService.Delete(id));
        }
    }
}
